sources/System/Lobby/ResumeLobbySystem.cpp

```diff
@@ -0,0 +1,226 @@
+		if (view.size() == 0)
+			return;
+		auto *texture = dynamic_cast<RAY::Texture *>(view.front().get<Drawable2DComponent>().drawable.get());
+		if (!texture)
+			return;
+		if (playersAreReady(*this->_wal.getScene()))
+			texture->setColor(WHITE);
+		else
+			texture->setColor(GRAY);
+	}
+
+	bool ResumeLobbySystem::playersAreReady(WAL::Scene &scene)
+	{
+		auto &lobby = scene.view<ResumeLobbyComponent>();
+		return std::all_of(lobby.begin(), lobby.end(), [](WAL::ViewEntity<ResumeLobbyComponent> &entity) {
+			auto &lobbyPlayer = entity.get<ResumeLobbyComponent>();
+			return lobbyPlayer.ready && lobbyPlayer.layout != ControllableComponent::NONE;
+		});
+	}
+
+	void ResumeLobbySystem::addController(WAL::Entity &player, ControllableComponent::Layout layout)
+	{
+		switch (layout) {
+			case ControllableComponent::KEYBOARD_0:
+			case ControllableComponent::KEYBOARD_1:
+				player.addComponent<KeyboardComponent>(layout);
+				break;
+			case ControllableComponent::GAMEPAD_0:
+				player.addComponent<GamepadComponent>(0);
+				break;
+			case ControllableComponent::GAMEPAD_1:
+				player.addComponent<GamepadComponent>(1);
+				break;
+			case ControllableComponent::GAMEPAD_2:
+				player.addComponent<GamepadComponent>(2);
+				break;
+			case ControllableComponent::GAMEPAD_3:
+				player.addComponent<GamepadComponent>(3);
+				break;
+			case ControllableComponent::AI:
+				player.addComponent<IAControllableComponent>("./assets/ai_scripts/john.lua");
+				break;
+			default:
+				throw std::runtime_error("Invalid controller for a player.");
+		}
+	}
+
+	void ResumeLobbySystem::resumeToGame(WAL::Wal &wal)
+	{
+		auto scene = Runner::gameState._loadedScenes[GameState::SceneID::GameScene];
+		int countPlayer = 0;
+
+		for (auto &[_, lobby] : wal.getScene()->view<ResumeLobbyComponent>()) {
+			if (lobby.layout == ControllableComponent::NONE)
+				continue;
+			auto &player = Runner::createPlayer(*scene);
+			player.setName(ParserYAML::playersInfos[countPlayer].name);
+			auto *position = player.tryGetComponent<PositionComponent>();
+			auto *bombHolder = player.tryGetComponent<BombHolderComponent>();
+			auto *model = player.tryGetComponent<Drawable3DComponent>();
+			auto *speed = player.tryGetComponent<SpeedComponent>();
+			if (position && bombHolder && model && speed) {
+				dynamic_cast<RAY3D::Model *>(model->drawable.get())->setTextureToMaterial(MAP_DIFFUSE,
+				                                                                          ParserYAML::playersInfos[countPlayer].asset);
+				position->position = ParserYAML::playersInfos[countPlayer].position;
+				bombHolder->explosionRadius = ParserYAML::playersInfos[countPlayer].explosionRange;
+				bombHolder->maxBombCount = ParserYAML::playersInfos[countPlayer].maxBombCount;
+				speed->speed = ParserYAML::playersInfos[countPlayer].speed;
+			}
+			addController(player, lobby.layout);
+			std::string texturePath = "assets/player/ui/" + _colors[lobby.playerColor] + ".png";
```

Comment:
You should create a static method in the LobbySystem that does what you copied here. Don't copy blocks of code when you could simply have a method.